File: website/code/entities/CalendarDate.ts
class CalendarDate {
    public year : number;
    public month : number;
    public day : number;

    constructor(year: number | string, month: number | string, day: number | string) {
        console.log('year: ' + year + ', month: ' + month + ', day: ' + day);

        this.year  = (typeof year === 'number')? year : parseInt(year);
        this.month = (typeof month === 'number')? month : parseInt(month);
        this.day   = (typeof day === 'number')? day : parseInt(day);
    }


    public toString(): string {
        return `${this.month}/${this.day}/${this.year}`;
    }
}

File: serverv2/APIs/Entities/CalendarDate.cs
using System;

namespace VideoGameArchive.Entities
{
    public class CalendarDate
    {
        public int year { get; set; }
        public int month { get; set; }
        public int day { get; set; }

        public CalendarDate(string date)
        {
            try {
                year = int.Parse(date.Substring(0, 4));
                month = int.Parse(date.Substring(4, 2));
                day = int.Parse(date.Substring(6));
            }
            catch(Exception ex) {
                Console.WriteLine($"error parsing date {date}: {ex.Message}");
            }
        }

        public string ToDateString()
        {
            return $"{month}/{day}/{year}";
        }

        public string ToUrlString()
        {
            return ToNumber().ToString();
        }

        public int ToNumber()
        {
            return year * 10000 + month * 100 + day;
        }

    }
}